Cql/Cql.Runtime/ValueSets/CqlCodeExtensions.cs

```diff
@@ -9,44 +9,59 @@
 
-        public static IEnumerable<CqlCode> Union(this IEnumerable<CqlCode> left, IEnumerable<CqlCode> right)
-        {
-            var l = left is IValueSetFacade facl ? facl : new InMemoryValueSet(left);
-            var r = right is IValueSetFacade facr ? facr : new InMemoryValueSet(right);
+    /// <summary>
+    /// Returns a facade on a <see cref="IValueSetDictionary"/> that represents access to the valueset within the dictionary.
+    /// </summary>
+    /// <remarks>This method does not check that the dictionary contains the given valueset.</remarks>
+    public static IValueSetFacade GetValueSet(this IValueSetDictionary all, CqlValueSet valueset) =>
+        GetValueSet(all, valueset.id ?? throw new ArgumentException("Valueset does not have a canonical."));
 
-            return l.Union(r);
-        }
 
-        public static IValueSetFacade Union(this IValueSetFacade left, IValueSetFacade right) => new ValueSetUnion(left, right);
+    /// <summary>
+    /// Create a union from two sets of codes.
+    /// </summary>
+    public static IEnumerable<CqlCode> Union(this IEnumerable<CqlCode> left, IEnumerable<CqlCode> right)
+    {
+        var l = left as IValueSetFacade ?? new InMemoryValueSet(left);
+        var r = right as IValueSetFacade ?? new InMemoryValueSet(right);
 
-        public static IEqualityComparer<CqlCode> ToEqualityComparer(this ICqlComparer comparer) =>
-            new CqlCodeEqualityHasher(comparer ?? throw new ArgumentNullException(nameof(comparer)));
+        return l.Union(r);
+    }
 
-        private class CqlCodeEqualityHasher : IEqualityComparer<CqlCode>
-        {
-            public ICqlComparer Comparer { get; }
+    /// <summary>
+    /// Create a union from two <see cref="IValueSetFacade"/>s.
+    /// </summary>
+    public static IValueSetFacade Union(this IValueSetFacade left, IValueSetFacade right) => new ValueSetUnion(left, right);
 
-            public CqlCodeEqualityHasher(ICqlComparer comparer)
-            {
-                Comparer = comparer ?? throw new ArgumentNullException(nameof(comparer));
-            }
+    /// <summary>
+    /// Create an <see cref="IEqualityComparer"/> based on the given <see cref="ICqlComparer"/>.
+    /// </summary>
+    public static IEqualityComparer<T> ToEqualityComparer<T>(this ICqlComparer<T> comparer, string? precision = null, bool useEquivalence = false) =>
+        new CqlCodeEqualityComparer<T>(comparer ?? throw new ArgumentNullException(nameof(comparer)), precision, useEquivalence);
 
-            public bool Equals(CqlCode? x, CqlCode? y) => Comparer.Compare(x, y, null) == 0;
+    private class CqlCodeEqualityComparer<T>
+        (ICqlComparer<T> comparer, string? precision = null, bool useEquivalence = false)
+        : IEqualityComparer<T>
+    {
+        public ICqlComparer<T> Comparer { get; } = comparer ?? throw new ArgumentNullException(nameof(comparer));
 
-            public int GetHashCode(CqlCode obj) =>
```

Comment:
So, we have a Comparer for equality, but for hashing the original code uses something else. This is bound to give problems in HashSets, since its lookup assumes that when x.Equals(y) then x.GetHashCode() == y.GetHashCode().